### DISCUSSION TITLE
Why does the project initialize the `static ggml_backend_registry reg;` in `llama_supports_rpc()`

### DISCUSSION THREAD

**Discussion by Hlinbit** [ggml-org/llama.cpp#10447] [Q&A] [answered]
I found a strange mechanism that `static ggml_backend_registry reg;` is initialized in `llama_supports_rpc()`. This is found when I debug the llama-cli. The function `llama_supports_rpc` is called in file `llama.cpp/common/arg.cpp`.

**Comment by slaren** [comment, marked as answer]
The backend registry is initialized the first time it is used, and `llama_supports_rpc` just happens to be the first function that is called that uses the backend registry. Why did you find it strange?

**Reply by Hlinbit** [reply]
You are right! I checked the whole function `common_params_parser_init` and found that `llama_supports_rpc` is the first function using the variable `static ggml_backend_registry reg`. The usage initialized the variable. This is not intentional, but it happens that `reg` is used here for the first time.
